pkgs/by-name/yo/yourkit-java/package.nix

```diff
@@ -0,0 +1,92 @@
+{ fetchzip
+, lib
+, stdenv
+, copyDesktopItems
+, imagemagick
+, makeDesktopItem
+, jre
+}:
+let
+  vPath = v: lib.elemAt (lib.splitString "-" v) 0;
+
+  version = "2024.3-b154";
+
+  arches = {
+    aarch64-linux = "arm64";
+    x86_64-linux = "x64";
+  };
+
+  arch = arches.${stdenv.targetPlatform.system} or (throw "Unsupported system");
+
+  hashes = {
+    arm64 = "sha256-r3DZtiuFhQcnkxhhm8mPibisGvDxn9AgNdX9S5HY9I4=";
+    x64 = "sha256-FtdukBezXXbFK+jLPjGdMGzesmWff1UKjev2JDIjP9M=";
+  };
+
+  desktopItem = makeDesktopItem {
+    name = "YourKit Java Profiler";
+    desktopName = "YourKit Java Profiler " + version;
+    type = "Application";
+    exec = "yourkit-java-profiler %f";
+    icon = "yourkit-java-profiler";
+    categories = [ "Development" "Java" "Profiling" ];
+    terminal = false;
+    startupWMClass = "YourKit Java Profiler";
+  };
+in
+stdenv.mkDerivation {
```

Comment:
Does this need a compiler or can it use `stdenvNoCC`?

Reply:
`stdenvNoCC` works.  All other suggestions have been incorporated too.